### DISCUSSION TITLE
Adapting your setup

### DISCUSSION THREAD

**Discussion by aranw** [malob/nix-config#14] [Q&A]
Hey 👋🏻 

I really like your current setup. I'm pretty new to Nix and have it working on a 2021 Macbook Pro with Apple Silicon and I'm looking to take my setup a little further with Flakes. I really like the approach you've got but the bit that is causing me a few issues when it comes to understanding it and adapting it is the cachix stuff. Is it possible for me to take your setup but not use the cachix stuff as I don't really particular want to invest in that service just yet (maybe in the future)

**Comment by aranw** [comment]
Just wanted to put an update to say. Thanks for this repo it's been really useful along with a video by @jmatsushita and following his posts and gists 👍🏻 

I think I've figured out what to do to remove cachix and it's now been removed. Not sure if I'll be able to get the same auto updating lock file that you've got but I'll try I think

**Reply by malob** [reply]
Hey @aranw, sorry for the late reply.

Using Cachix is not required. There are two places I use Cachix in my configs, as an additional binary cache that the config will pull from, and it the GitHub Workflow that I use for CI and automatically updating my inputs once a week.

For the former, just remove references to my Cachix binary cache in the `nix.binaryCaches` and `nix.binaryCachePublicKeys` options, which are located in `darwin/bootstrap.nix` in my config:
https://github.com/malob/nixpkgs/blob/d20f0a1bc0ac5f4c49883196883a86f56874cccc/darwin/bootstrap.nix#L6-L13

For the latter, I think all you'll need to do is remove the "Setup Cachix" steps in `.github/workflows/ci.yml`:
https://github.com/malob/nixpkgs/blob/d20f0a1bc0ac5f4c49883196883a86f56874cccc/.github/workflows/ci.yml#L32-L36
https://github.com/malob/nixpkgs/blob/d20f0a1bc0ac5f4c49883196883a86f56874cccc/.github/workflows/ci.yml#L80-L84
